packages/core/src/Menu/Menu.js

```diff
@@ -12,54 +12,89 @@ import { MenuList } from '../MenuList/MenuList.js'
  * @param {Menu.PropTypes}
  * @returns {React.Component}
  *
- * @example import { Menu } from '@dhis2/ui-core'
+ * @example import { Menu } from '@dhis2/ui'
  *
  * @see Specification: {@link https://github.com/dhis2/design-system/blob/master/molecules/menu.md|Design system}
- * @see Live demo: {@link /demo/?path=/story/menu--default|Storybook}
+ * @see Live demo: {@link /demo/?path=/story/components-core-menu--default|Storybook}
  */
-const Menu = ({ children, className, dataTest, maxWidth }) => (
-    <div className={className} data-test={dataTest}>
-        <Card>
-            <div className="menu-list-wrapper">
-                <MenuList>{children}</MenuList>
-            </div>
-        </Card>
+const Menu = ({
+    children,
+    className,
+    dataTest,
+    dense,
+    maxHeight,
+    maxWidth,
+}) => {
+    const [openedSubMenu, setOpenedSubMenu] = useState(null)
+    const toggleSubMenu = index => {
+        const toggleValue = index === openedSubMenu ? null : index
+        setOpenedSubMenu(toggleValue)
+    }
+
+    const cardCSS = resolve`
+        padding: ${spacers.dp4} 0;
+        max-height: ${maxHeight};
+        overflow-y: auto;
+    `
 
-        <style jsx>{`
-            .menu-list-wrapper {
-                padding: ${spacers.dp4} 0;
-            }
-            div {
-                display: inline-block;
-                min-width: 128px;
-            }
-        `}</style>
-        <style jsx>{`
-            div {
-                max-width: ${maxWidth};
-            }
-        `}</style>
-    </div>
-)
+    return (
+        <div className={className} data-test={dataTest}>
+            <Card className={cardCSS.className}>
+                <MenuList>
+                    {Children.map(children, (child, index) =>
+                        isValidElement(child)
+                            ? cloneElement(child, {
+                                  showSubMenu: openedSubMenu === index,
+                                  toggleSubMenu: toggleSubMenu.bind(
+                                      this,
+                                      index
+                                  ),
+                                  dense:
+                                      typeof child.props.dense === 'boolean'
+                                          ? child.props.dense
+                                          : dense,
+                              })
+                            : child
+                    )}
+                </MenuList>
+            </Card>
+
+            <style jsx>{`
+                div {
+                    display: inline-block;
+                    min-width: 128px;
```

Comment:
This should be over-writable as well. If I want to use the menu in the sidebar, I'd want it to always take 100% of the space. So either we add a `useFullWidth` prop (just an ad hoc name) or we allow `min-width` to be changed to `100%`.

Reply:
With the select we had a similar issue. We decided to always render full width there. For consistency I think it'd be nice to do the same here. Then we'd only set the width for the popups (as those are independent of the layout).

Reply:
In the `MenuList` demo stories I've included an example for creating a sidebar menu. The point is, you'd use the `MenuList` for that, and not the `Menu`. The `MenuList` just occupies the full width of its parent.

To me this makes sense, and I even think that is why @Mohammer5 extracted the `MenuList` from the `Menu` a long time ago. See https://github.com/dhis2/ui-core/pull/297

Reply:
> `min-width: 128px;`
> This should be over-writable as well. 

I was actually going to point you to the [design specs](https://www.sketch.com/s/07b991f1-a188-4977-91af-40c51f24ad0a/a/DQAppk) to show you that these mention a fixed `min-width`. Not sure if that has been changed or if I just got confused. I'll check with @cooper-joe

Reply:
Aha, yes there are the fixed sizes, they were slightly hidden in a different place they used to be, because dense and normal are now slightly different:
![image (2)](https://user-images.githubusercontent.com/353236/82448152-a369ec80-9aa9-11ea-8627-342ee5518ffa.png)
![image (1)](https://user-images.githubusercontent.com/353236/82448168-a7960a00-9aa9-11ea-8347-653c11fcc398.png)

It used to be a single fixed min-width, but now dense and normal have their own. FIxed that in 7ae718a.

Reply:
> and I even think that is why @Mohammer5 extracted the MenuList from the Menu a long time ago

I don't remember that :rofl: 

> In the MenuList demo stories I've included an example for creating a sidebar menu. The point is, you'd use the MenuList for that, and not the Menu. The MenuList just occupies the full width of its parent.

I think we have to work on the names then as they don't convey their intended usecases. Terminology-wise, I'd expect to use a `Menu`, when I need a `Menu`. If I had to use a `MenuList` as a menu, I'd expect it to be a list of menus. It does make sense to have a `MenuList` as an internal component so we wrap everything with an `ul`, but once it becomes public or even has different use-cases, then this is a rather confusing name.

So the `Menu` is actually a `PoppableMenu` and the `MenuList` is a `MenuItemContainer`. These are just ad hoc names, not meant as serious names, but they illustrate what I want to say: They convey a very clear intended usage, while `Menu` and `MenuList` do not.